test/unit/executionPlanWebviewController.test.ts

```diff
@@ -0,0 +1,554 @@
+
+        mockExecutionPlanService =
+            sandbox.createStubInstance(ExecutionPlanService);
+        mockUntitledSqlDocumentService = sandbox.createStubInstance(
+            UntitledSqlDocumentService,
+        );
+
+        controller = new ExecutionPlanWebviewController(
+            mockContext,
+            mockExecutionPlanService,
+            mockUntitledSqlDocumentService,
+            executionPlanContents,
+            xmlPlanFileName,
+        );
+    });
+
+    teardown(() => {
+        sandbox.restore();
+    });
+
+    test("should initialize with correct state and webview title", () => {
+        const initialState: ep.ExecutionPlanWebviewState = {
+            executionPlanState: {
+                loadState: ApiStatus.Loading,
+                executionPlanGraphs: [],
+                totalCost: 0,
+            },
+        };
+
+        assert.deepStrictEqual(
+            controller.state,
+            initialState,
+            "Initial state should match",
+        );
+        assert.deepStrictEqual(
+            controller.panel.title,
+            xmlPlanFileName,
+            "Webview Title should match",
+        );
+    });
+
+    test("should call createExecutionPlanGraphs in getExecutionPlan reducer", async () => {
+        // Stub createExecutionPlanGraphs to mock its behavior
+        const createExecutionPlanGraphsStub = sandbox
+            .stub(epUtils, "createExecutionPlanGraphs")
+            .resolves({
+                executionPlanState: {
+                    executionPlanGraphs: [],
+                    loadState: ApiStatus.Loaded,
+                    totalCost: 100,
+                },
+            });
+
+        // Mock state and payload for the reducer
+        const mockState = {
+            executionPlanState: {
+                loadState: ApiStatus.Loading,
+                executionPlanGraphs: [],
+                totalCost: 0,
+            },
+        };
+
+        await controller["_reducers"]["getExecutionPlan"](mockState, {});
+
+        assert.ok(
+            createExecutionPlanGraphsStub.calledOnce,
+            "createExecutionPlanGraphs should be called once",
+        );
+
+        assert.deepStrictEqual(
+            createExecutionPlanGraphsStub.firstCall.args,
+            [
+                mockState,
+                controller.executionPlanService,
+                [controller.executionPlanContents],
+            ],
+            "createExecutionPlanGraphs should be called with correct arguments",
+        );
+
+        createExecutionPlanGraphsStub.restore();
+    });
+
+    test("should call saveExecutionPlan in saveExecutionPlan reducer", async () => {
+        const saveExecutionPlanStub = sandbox
+            .stub(epUtils, "saveExecutionPlan")
+            .resolves({
+                executionPlanState: {
+                    executionPlanGraphs: [],
+                    loadState: ApiStatus.Loaded,
+                    totalCost: 100,

```

Comment:
If there's some state that's shared across tests, you can pull it out into a shared const.  If the shape of the state changes in the future, it helps to minimize the number of spots that need updating